e2e/interchaintestv8/chainconfig/encoding.go

```diff
@@ -30,53 +41,72 @@ import (
 	ibctmtypes "github.com/cosmos/ibc-go/v9/modules/light-clients/07-tendermint"
 )
 
-// Codec returns the global E2E protobuf codec.
-func Codec() *codec.ProtoCodec {
-	cdc, _ := codecAndEncodingConfig()
-	return cdc
+// CosmosEncodingConfig returns the global E2E encoding config for simd.
+func CosmosEncodingConfig() *sdktestutil.TestEncodingConfig {
+	return encodingConfig("cosmos")
 }
 
-// SDKEncodingConfig returns the global E2E encoding config.
-func SDKEncodingConfig() *sdktestutil.TestEncodingConfig {
-	_, cfg := codecAndEncodingConfig()
-	return &sdktestutil.TestEncodingConfig{
-		InterfaceRegistry: cfg.InterfaceRegistry,
-		Codec:             cfg.Codec,
-		TxConfig:          cfg.TxConfig,
-		Amino:             cfg.Amino,
+// EncodingConfig returns the global E2E encoding config.
+// It includes CosmosSDK, IBC, and Wasm messages
+func encodingConfig(bech32Prefix string) *sdktestutil.TestEncodingConfig {
+	amino := codec.NewLegacyAmino()
+	interfaceRegistry, err := codectypes.NewInterfaceRegistryWithOptions(codectypes.InterfaceRegistryOptions{
+		ProtoFiles: proto.HybridResolver,
+		SigningOptions: txsigning.Options{
+			AddressCodec: address.Bech32Codec{
+				Bech32Prefix: bech32Prefix,
+			},
+			ValidatorAddressCodec: address.Bech32Codec{
+				Bech32Prefix: bech32Prefix + sdk.PrefixValidator + sdk.PrefixOperator,
+			},
+		},
+	})
+	if err != nil {
+		panic(err)
 	}
-}
```

Comment:
SDK52 way was better. We can't keep?

Reply:
Turns out it was no even SDK52. This was possible in 50 :P Adding it back